Doc/c-api/hash.rst

```diff
@@ -41,8 +45,23 @@ See also the :c:member:`PyTypeObject.tp_hash` member.
    .. versionadded:: 3.4
 
 
+Functions
+^^^^^^^^^
+
+.. c:function:: Py_hash_t PyHash_Double(double value)
```

Comment:
I would prefer to expose this as unstable API.  Hashing of numeric types is relatively low-level detail of implementation, which was changed in past for minor (3.x.0) releases (nans hashing was last).  Why not keep this freedom in future?

Reply:
I don't expect PyHash_Double() **API** to change in the future. The *result* of the function can change in Python 3.x.0 releases, but I don't consider that it qualifies the function for the PyUnstable API.

The PyUnstable API is more when there is a risk that the function can be removed, that its API can change, or that a major change can happen in a Python 3.x.0 release.

In Python 3.2 (2010), `_Py_HashChange()` was written in commit dc787d2055a7b562b64ca91b8f1af6d49fa39f1c of issue gh-52435.

```
commit dc787d2055a7b562b64ca91b8f1af6d49fa39f1c
Author: Mark Dickinson <dickinsm@gmail.com>
Date:   Sun May 23 13:33:13 2010 +0000

    Issue #8188: Introduce a new scheme for computing hashes of numbers
    (instances of int, float, complex, decimal.Decimal and
    fractions.Fraction) that makes it easy to maintain the invariant that
    hash(x) == hash(y) whenever x and y have equal value.
```

As written above, the latest major change was in Python 3.10 to treat NaN differently.

Reply:
> The result of the function can change in Python 3.x.0 releases

Previously, the function signature for ``_Py_HashDouble()`` was changed too.